warehouse/query-core/src/main/java/datawave/query/transformer/ExcerptTransform.java

```diff
@@ -319,51 +337,70 @@ private Set<Excerpt> getExcerpts(PhraseIndexes phraseIndexes) {
             } else {
-                hitPhrase.add(phrasePart);
+                // set the end of the range
+                excerptIteratorOptions.put(TermFrequencyExcerptIterator.END_OFFSET, String.valueOf(end + expandSize));
             }
-        }
+            if (log.isDebugEnabled() && i != 0) {
+                log.debug("size of excerpt requested: {}", excerptFields.getOffset(field) * 2);
+                log.debug("original range is ({},{}) and the expanded range is ({},{})", start, end,
+                                excerptIteratorOptions.get(TermFrequencyExcerptIterator.START_OFFSET),
+                                excerptIteratorOptions.get(TermFrequencyExcerptIterator.END_OFFSET));
+            }
+            try {
+                excerptIterator.init(source, excerptIteratorOptions, env);
+                excerptIterator.setHitTermsList(hitTermValues);
+                excerptIterator.setDirection(excerptFields.getDirection(field).toUpperCase().trim());
+                excerptIterator.setOrigHalfSize(origHalfSize);
+                if (i != 0) {
+                    excerptIterator.setTrim(true);
+                }
+                excerptIterator.seek(range, Collections.emptyList(), false);
+                // if an excerpt is returned...
+                if (excerptIterator.hasTop()) {
+                    Key topKey = excerptIterator.getTopKey();
+                    String[] parts = topKey.getColumnQualifier().toString().split(Constants.NULL);
+                    // The column qualifier is expected to be field\0phraseWithScores\0phraseWithoutScores\0oneBestExcerpt.
+                    if (parts.length != 4) {
+                        log.warn("{} returned top key with incorrectly-formatted column qualifier in key: {} when scanning for excerpt [{},{}] for field {} within range {}",
+                                        TermFrequencyExcerptIterator.class.getSimpleName(), topKey, start, end, field, range);
+                        return null;
+                    }
 
-        // return phrase based on direction
-        String result = String.join(" ", hitPhrase); // if no direction given, return everything
-        String direction = excerptFields.getDirection(phraseParts[0]).toUpperCase().trim();
-        if (direction.equals(BEFORE)) { // remove tokens prior to hit term
-            result = StringUtils.substringBeforeLast(result, "]") + "]";
-        } else if (direction.equals(AFTER)) { // remove tokens after hit term
-            result = "[" + StringUtils.substringAfter(result, "[");
+                    // if we have reached the limit of times to try, or we have no stop words removed
+                    if (i == 1 || !parts[1].equals("XXXWESKIPPEDAWORDXXX")) {
+                        return parts[1] + Constants.NULL + parts[2] + Constants.NULL + parts[3];
+                    }
+                } else {
+                    return null;
```

Comment:
Would we possibly want to still make the second attempt in this case if we're only on the first? Consider calling `continue` or `break` instead?

Reply:
I guess it wouldn't hurt to try. But I think if hasTop is still false after the second attempt we might want to throw/log an error as something probably went terribly wrong if this happens.